Add support for infinite auto-restarting of KafkaConnect connectors (#9041)
strimzi/strimzi-kafka-operator: documentation/api/io.strimzi.api.kafka.model.AutoRestart.adoc

Configures automatic restarts for connectors and tasks that are in a `FAILED` state.

When enabled, a back-off algorithm applies the automatic restart to each failed connector and its tasks.

> The back-off is calculated using the formula `n * n + n` where `n` is number of previous restarts and is capped at 60 minutes as a maximum.

Reviewer: ```suggestion
An incremental back-off interval is calculated using the formula `n * n + n` where `n` represents the number of previous restarts. 
This interval is capped at a maximum of 60 minutes.
```

> That means that the restarts will be done immediately and then after 2, 6, 12, 20, 30, 42, 56, and then every 60 minutes.

Reviewer: ```suggestion
Consequently, a restart occurs immediately, followed by restarts after 2, 6, 12, 20, 30, 42, 56 minutes, and then at 60-minute intervals.
```

> By default, Strimzi will restart the connector and its tasks forever.

Reviewer: ```suggestion
By default, Strimzi initiates restarts of the connector and its tasks indefinitely.
```

> By default, Strimzi will be restarting the connector and its tasks forever.

Reviewer 2: ```suggestion
By default, Strimzi will restart the connector and its tasks forever.
```

> If you want, you can use the `maxRestarts` field to set the maximum number of restarts.

Reviewer: ```suggestion
However, you can use the `maxRestarts` property to set a maximum on the number of restarts.
```

> When `maxRestarts` are used and the connector fails even after the final attempt, you will have to restart it manually.

Reviewer: ```suggestion
If `maxRestarts` is configured and the connector still fails even after the final restart attempt, you must then restart the connector manually.
```

    enabled: true
----

> If desired, you can also set the maximum number of restarts.

Reviewer: ```suggestion
If you prefer, you can also set a maximum limit on the number of restarts.
```
